File: server/fastapi/config/routes/30_figma.routes.py
"""GET /healthz/integrations/figma/me — proxy of upstream /v1/me."""

from __future__ import annotations

import os

from fastapi import APIRouter, Depends, Response, status
from fetch_http_client import AsyncClient

from ._di import get_figma_client


def mount(_app, _config) -> APIRouter:
    router = APIRouter()

    @router.get("/healthz/integrations/figma/me")
    async def figma_me(
        response: Response,
        client: AsyncClient = Depends(get_figma_client),
    ):
        try:
            upstream = await client.get("/v1/me")
        except Exception as exc:  # noqa: BLE001
            response.status_code = status.HTTP_502_BAD_GATEWAY
            return {"service": "figma", "connected": False, "error": str(exc)}
        if upstream.status_code >= 400:
            response.status_code = status.HTTP_502_BAD_GATEWAY
            return {
                "service": "figma",
                "connected": False,
                "upstream_status": upstream.status_code,
                "upstream_body": upstream.text[:512] if hasattr(upstream, "text") else "",
            }
        return {
            "service": "figma",
            "connected": True,
            "host": os.environ.get("FIGMA_API_BASE_URL", "https://api.figma.com"),
            "data": upstream.json(),
        }

    return router

File: server/fastify/config/routes/30_figma.routes.mjs
export default async function figmaRoutes(fastify, _config) {
  fastify.get("/healthz/integrations/figma/me", async (request, reply) => {
    const client = await request.fetchClient("figma");
    try {
      const upstream = await client.get("/v1/me");
      if (upstream.status_code >= 400) {
        const body = await upstream.text();
        return reply.code(502).send({
          service: "figma",
          connected: false,
          upstream_status: upstream.status_code,
          upstream_body: body.slice(0, 512),
        });
      }
      const data = await upstream.json();
      return {
        service: "figma",
        connected: true,
        host: process.env.FIGMA_API_BASE_URL ?? "https://api.figma.com",
        data,
      };
    } catch (err) {
      return reply.code(502).send({ service: "figma", connected: false, error: err.message });
    }
  });
}
